zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -1397,7 +1397,7 @@ private void checkpointNote(NotebookSocket conn, Notebook notebook,
     String commitMessage = (String) fromMessage.get("commitMessage");
     AuthenticationInfo subject = new AuthenticationInfo(fromMessage.principal);
     Revision revision = notebook.checkpointNote(noteId, commitMessage, subject);
-    if (revision != null) {
+    if (!Revision.EMPTY.equals(revision)) {
```

Comment:
maybe we can come up with a better implementation here like `revision.isEmpty()` or `Revision.isEmpty(revision)`

what do you think